### PR TITLE
Add e2e tests for consensus-service

### LINKED ISSUE #6020: Add e2e tests for consensus-service
Ensure EET's exist to update and remove:
 - [ ] Topic auto-renew accounts.
 - [ ] Topic `submitKey`s.
 - [ ] Topic `adminKeys`s. 
 - [ ] Replaced TopicMetadata to Topic
 - [ ] Refactored some testing classes
 - [ ] Checked through all the consensus service code and did some updates with the latest changes.
 - [ ] Added deleting adminKey in ConsensusUpdateTopicHandler and the unit test.

### PR DESCRIPTION
**Pull request by iwsimon**
Replaced TopicMetadata with Topic
Refactored some testing classes,
Checked through all the consensus service code and did some updates with the latest changes.
Added deleting adminKey in ConsensusUpdateTopicHandler and the unit test.

**Description**:

**Related issue(s)**:

Fixes #6020 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by hendrikebbers — APPROVED** [review]

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java [hunk 1]
```diff
@@ -61,13 +61,13 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
...
         if (op.hasAdminKey()) {
-            context.requireKeyOrThrow(op.adminKeyOrThrow(), INVALID_ADMIN_KEY);
+            context.requireKeyOrThrow(op.adminKey(), INVALID_ADMIN_KEY);
         }
 
         // If an account is to be used for auto-renewal, then the account must exist and the transaction
         // must be signed with that account's key.
         if (op.hasAutoRenewAccount()) {
-            final var autoRenewAccountID = op.autoRenewAccountOrThrow();
+            final var autoRenewAccountID = op.autoRenewAccount();
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java:70]: Why are these changes getting rid of the `OrThrow`?

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java [hunk 2]
```diff
@@ -134,21 +136,20 @@ public Response findResponse(@NonNull final QueryContext context, @NonNull final
...
-            info.expirationTime(meta.expirationTimestamp());
+            info.expirationTime(Timestamp.newBuilder().seconds(meta.expiry()).build());
             if (!isEmpty(meta.adminKey())) info.adminKey(meta.adminKey());
             if (!isEmpty(meta.submitKey())) info.submitKey(meta.submitKey());
-            info.autoRenewPeriod(Duration.newBuilder().seconds(meta.autoRenewDurationSeconds()));
-            meta.autoRenewAccountId()
-                    .ifPresent(account ->
-                            info.autoRenewAccount(AccountID.newBuilder().accountNum(account)));
+            info.autoRenewPeriod(Duration.newBuilder().seconds(meta.autoRenewPeriod()));
+            if (meta.autoRenewAccountNumber() != 0)
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:151]: @Neeharika-Sompalli or @tinker-michaelj should this be
```suggestion
            if (meta.autoRenewAccountNumber() > 0)
```
?

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java [hunk 3]
```diff
@@ -111,14 +112,16 @@ public void handle(
 
         /* since we have validated topic exists, topic.get() is safe to be called */
         try {
+
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java:115]: extra line?

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java [hunk 4]
```diff
@@ -111,14 +112,16 @@ public void handle(
...
             final var updatedTopic = updateRunningHashAndSequenceNumber(txn, topic.get(), handleContext.consensusNow());
 
             /* --- Put the modified topic. It will be in underlying state's modifications map.
             It will not be committed to state until commit is called on the state.--- */
             topicStore.put(updatedTopic);
 
-            recordBuilder.setNewTopicMetadata(
+            recordBuilder.setNewTopic(
                     asBytes(updatedTopic.runningHash()), updatedTopic.sequenceNumber(), RUNNING_HASH_VERSION);
+
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java:124]: extra line?

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java [hunk 5]
```diff
@@ -135,7 +137,12 @@ private void resolveMutableBuilderAttributes(
             @NonNull final Topic.Builder builder,
             @NonNull final Topic topic) {
         if (op.hasAdminKey()) {
-            builder.adminKey(op.adminKey());
+            var key = op.adminKey();
+            if (isEmpty(key)) {
+                builder.adminKey((Key) null);
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:142]: Why is this needed?

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusCreateTopicHandlerTest.java [hunk 6]
```diff
@@ -308,9 +301,9 @@ void translatesInvalidExpiryException() {
         given(expiryValidator.resolveCreationAttempt(anyBoolean(), any()))
                 .willThrow(new HandleException(ResponseCodeEnum.INVALID_EXPIRATION_TIME));
 
-        final var failure = assertThrows(
+        final var msg = assertThrows(
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusCreateTopicHandlerTest.java:304]: Not sure why this was renamed to `msg` instead of `failure`. The `assertThrows` method returns an exception, not a string/message

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusDeleteTopicHandlerParityTest.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Mock
+    private ReadableTopicStore mockStore;
+
+    private ConsensusDeleteTopicHandler subject;
+
+    @BeforeEach
+    void setUp() {
+        subject = new ConsensusDeleteTopicHandler();
+        mockStore = mock(ReadableTopicStore.class);
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusDeleteTopicHandlerParityTest.java:55]: You don't need this line if you annotate the class with `@ExtendWith(MockitoExtension.class)`

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusDeleteTopicHandlerParityTest.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        accountStore = AdapterUtils.wellKnownKeyLookupAt();
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusDeleteTopicHandlerParityTest.java:56]: Why is `accountStore` annotated with `@Mock` _and_ reassigned here?

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java [hunk 9]
```diff
@@ -74,6 +75,7 @@ void setUp() {
     }
 
     @Test
+    @DisplayName("extractHeader as expected")
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java:78]: There's no value added with this and a few other of the display names. It's just extra code to read that duplicates what's already in the method name

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java [hunk 10]
```diff
@@ -139,10 +145,28 @@ void validatesQueryIfInvalidTopic() throws Throwable {
     }
 
     @Test
+    @DisplayName("Topic Id in transaction is needed during validate")
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java:148]: This display name is much more useful! 👍

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusSubmitMessageHandlerTest.java [hunk 11]
```diff
@@ -340,11 +273,11 @@ void failsIfChunkTxnPayerIsNotInitialPayer() {
     }
 
     @Test
-    @DisplayName("Handle fails if submit message chunk txn payer is not same as initial txn payer")
+    @DisplayName("Handle fails if submit message chunk txn id is not same as initial txn id")
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusSubmitMessageHandlerTest.java:276]: This change makes the comment less clear IMO

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusUpdateTopicHandlerTest.java [hunk 12]
```diff
@@ -329,64 +351,67 @@ void appliesNewAutoRenewNumViaMeta() {
...
         assertEquals(topic, newTopic);
     }
 
-    public static void assertFailsWith(final ResponseCodeEnum status, final Runnable something) {
-        final var ex = assertThrows(HandleException.class, something::run);
-        assertEquals(status, ex.getStatus());
-    }
-
     @Test
+    @DisplayName("Validate Mutate NonExpiryField memo as expected")
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusUpdateTopicHandlerTest.java:370]: Quite a few more display names here that don't add any information to the person reading the code

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java (hunk 1 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java:70]: I use the next line, context.requireKeyOrThrow(autoRenewAccountID, INVALID_AUTORENEW_ACCOUNT) to capture the PreCheckException and handle it to workflow to handle.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java (hunk 2 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:151]: The builder default autoRenewAccountNumber is 0. I will also let @Neeharika-Sompalli and @tinker-michaelj answer this.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java (hunk 3 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java]: removed, applied spotless.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java (hunk 4 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java]: removed, thanks.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java (hunk 5 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:142]: If empty keyList - the adminKey is cleared. This is how to remove the adminKey.

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusCreateTopicHandlerTest.java (hunk 6 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusCreateTopicHandlerTest.java:304]: Just to be consistent with all other code. msg or failure is just a var name.

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusDeleteTopicHandlerParityTest.java (hunk 7 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusDeleteTopicHandlerParityTest.java]: updated, thanks.

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java (hunk 9 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java:78]: Just to show which method is tested.

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java (hunk 10 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java:148]: thanks.

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusSubmitMessageHandlerTest.java (hunk 11 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusSubmitMessageHandlerTest.java:276]: two tests are next each other, one is on txn payer, another is on txn id.

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusUpdateTopicHandlerTest.java (hunk 12 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusUpdateTopicHandlerTest.java:370]: This is just to test one of || conditions in wantsToMutateNonExpiryField(). It's hard to put a real meaningful name.

File: hedera-node/hedera-consensus-service-impl/build.gradle.kts [hunk 13]
```diff
@@ -28,6 +28,8 @@ configurations.all {
 }
 
 dependencies {
+  implementation(project(mapOf("path" to ":hedera-node:hedera-config")))
+  implementation(project(mapOf("path" to ":hedera-node:hedera-config")))
```
Comment (hendrikebbers) [inline hedera-node/hedera-consensus-service-impl/build.gradle.kts:32]:
```suggestion
  implementation(project(":hedera-node:hedera-config"))
```

File: hedera-node/hedera-consensus-service-impl/build.gradle.kts [hunk 14]
```diff
@@ -28,6 +28,8 @@ configurations.all {
 }
 
 dependencies {
+  implementation(project(mapOf("path" to ":hedera-node:hedera-config")))
```
Comment (hendrikebbers) [inline hedera-node/hedera-consensus-service-impl/build.gradle.kts:31]:
```suggestion
  implementation(project(":hedera-node:hedera-config"))
```

Comment (hendrikebbers) [reply hedera-node/hedera-consensus-service-impl/build.gradle.kts:31]: Is the `mapOf` added automatically when you let IntellIJ add the dependency or how is it created?

File: hedera-node/hedera-consensus-service-impl/build.gradle.kts (hunk 13 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/build.gradle.kts]: Updated, thanks.

File: hedera-node/hedera-consensus-service-impl/build.gradle.kts (hunk 14 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/build.gradle.kts]: IntelliJ added it. I also noticed IntelliJ added the duplicated line too.

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java (hunk 9 above)
Comment (mhess-swl) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java:78]: By default, junit already displays the method name. I would suggest either putting more info about the test in these messages, or just removing it

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusSubmitMessageHandlerTest.java (hunk 11 above)
Comment (mhess-swl) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusSubmitMessageHandlerTest.java:276]: ahh got it!

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java (hunk 5 above)
Comment (mhess-swl) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:142]: Ahh got it. I wasn't thinking this one through enough. Proceed!

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:151]: Since we validate the auto-renew account at the time it is set on the `Topic`, the only options here will be:
 1. `number = 0` (no auto-renew account); or,
 2. `number = X > 0` (where `0.0.X` was a valid auto-renew account at the time it was set on this `Topic`).

If you want, we could log at `ERROR` if the number is negative; but it should be impossible.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java [hunk 15]
```diff
@@ -135,7 +137,12 @@ private void resolveMutableBuilderAttributes(
             @NonNull final Topic.Builder builder,
             @NonNull final Topic topic) {
         if (op.hasAdminKey()) {
-            builder.adminKey(op.adminKey());
+            var key = op.adminKey();
+            if (isEmpty(key)) {
```
Comment (tinker-michaelj) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:141]: Hmm...we would like to _only_ make the topic immutable when the `op.adminKey()` is exactly an empty `KeyList`. (c.f. protobuf message comment [here](https://github.com/hashgraph/hedera-protobufs/blob/main/services/consensus_update_topic.proto#L58)).

On the other hand, it seems possible to use **any** empty `Key` to remove the admin key in `mono-service`; though the EET [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/consensus/TopicUpdateSuite.java#L173) does only test exactly the empty `KeyList`.

I'd suggest adding another EET like the one above that characterizes the current behavior when the empty key is for example, `ThresholdKey{2/2, keys=[KeyList{keys=[]}, KeyList{keys=[]}]}` and seeing what happens. 



I think it would make sense to add an `AttributeValidator.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java (hunk 2 above)
Comment (mhess-swl) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:151]: Ah got it, ok that works. Thanks!

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java (hunk 9 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java]: I updated the display. Please see if it makes more sense to you. I tried to make sure every unit test method has @DisplayName.

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusUpdateTopicHandlerTest.java (hunk 12 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusUpdateTopicHandlerTest.java]: I updated the display. Please see if it makes more sense to you. I tried to make sure every unit test method has @DisplayName.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java (hunk 15 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:141]: Thanks. I added two more unit tests to test EMPTY_KEYLIST and EMPTY_THRESHOLD_KEY(your mentioned above) in ConsensusUpdateTopicHandlerTest. I also added one test in TopicUpdateSuite to test the ThresholdKey you mentioned. Interestingly, in mono code, EMPTY_THRESHOLD_KEY can remove admin key; in module code, EMPTY_THRESHOLD_KEY can not remove admin key because it's not consider to be KeyUtils.isEmpty(). Is that a correct behavior?

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java (hunk 9 above)
Comment (mhess-swl) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusGetTopicInfoHandlerTest.java]: These look better, thank you! 😄

**Review by mhess-swl — APPROVED** [review]

**Review by povolev15 — APPROVED** [review]
LGTM

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java (hunk 15 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java]: Followed the discussion from today's modularization call, I updated the code to clear adminKey with only empty KeyList and their unit tests as well. ThresholdKey{2/2, keys=[KeyList{keys=[]}, KeyList{keys=[]}]} can't clear adminKey in module code. In mono code, if passing this Threholdkey in the transaction, it failed because "No sig control for it", so you can't use it to clear adminKey either.

Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java]: I added AttributeValidator.isImmutableKey(Key key). @tinker-michaelj, Please check if this conversation can be resolved. Thanks.

File: hedera-node/hedera-consensus-service-impl/build.gradle.kts (hunk 14 above)
Comment (hendrikebbers) [reply hedera-node/hedera-consensus-service-impl/build.gradle.kts]: it ids not only about the duplicate lines. We do not want to use `mapOf`.

Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/build.gradle.kts]: Those mapOfs have been removed too.

**Review by povolev15 — APPROVED** [review]

File: hedera-node/hedera-consensus-service-impl/build.gradle.kts (hunk 13 above)
Comment (hendrikebbers) [reply hedera-node/hedera-consensus-service-impl/build.gradle.kts]: Have you missed to push? I do not see any change (see attached screenshoot)
Bildschirmfoto 2023-05-19 um 14 07 21

**Review by hendrikebbers — APPROVED** [review]

**Review by povolev15 — APPROVED** [review]

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java (hunk 15 above)
Comment (iwsimon) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java]: I created a new issue#6690 following the #ask-an-expert question/answer regarding to delete adminKey/autoRenewAccount.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/StandardizedAttributeValidator.java [hunk 16]
```diff
@@ -136,4 +137,10 @@ private static boolean contains(byte[] a, byte val) {
         }
         return false;
     }
+
+    @Override
+    public boolean isImmutableKey(Key key) {
```
Comment (tinker-michaelj) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/StandardizedAttributeValidator.java:142]: Is this intended to be,
```
public boolean isImmutableKey(@NonNull final Key key) {
        return requireNonNull(key).hasKeyList() && key.keyList().isEmpty();
}
```
As written seems to have the null check in the wrong place; and `keysOrElse()` is redundant if `hasKeyList()` is true.

Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/StandardizedAttributeValidator.java:142]: Added requireNonNull to remove sonar warning: Method invocation 'keysOrElse may produce 'NullPointerException'.
Same for keysOrElse(emptyList()). key.keyList().isEmpty() is not valid for the hapi.node.base.Key

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Review by povolev15 — APPROVED** [review]
LGTM

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/StandardizedAttributeValidator.java [hunk 17]
```diff
@@ -136,4 +137,11 @@ private static boolean contains(byte[] a, byte val) {
         }
         return false;
     }
+
+    @Override
+    public boolean isImmutableKey(@NonNull Key key) {
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/StandardizedAttributeValidator.java:142]: This method name seems very unintuitive, since “immutable” describes the key, but in this case doesn’t describe the key, instead describing the immutable entity associated with it. A comment about what this actually means would be helpful. Even better would be to use a name that doesn't label the key "immutable" (when the key isn't actually immutable)

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/StandardizedAttributeValidator.java (hunk 17 above)
Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/StandardizedAttributeValidator.java:142]: There is a Java doc in interface AttributeValidator: Validates if immutable entity with the key and @return true if immutable entity with the key

**Review by JeffreyDallas — APPROVED** [review]

**Review by JeffreyDallas — APPROVED** [review]

**Review by JeffreyDallas — APPROVED** [review]
